daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/CidContainer.scala

```diff
@@ -74,10 +74,7 @@ trait CidContainer[+A] {
     suffixer.traverse[String] {
       case Value.ContractId.V1(discriminator, Bytes.Empty) =>
         Value.ContractId.V1.build(discriminator, f(discriminator))
-      case acoid @ Value.ContractId.V1(_, _) =>
-        Right(acoid)
-      case acoid @ Value.ContractId.V0(_) =>
-        Left(s"expect a Contract ID V1, found $acoid")
+      case acoid => Right(acoid)
```

Comment:
(minor) I prefer to keep the pattern exhaustiveness check.  

```suggestion
      case acoid: Value.ContractId.V0 => Right(acoid)
```